test/suite/test_truncate25.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+from wtdataset import SimpleDataSet
+
+# test_truncate25.py
+# Ensure that any history store records are correctly removed when doing
+# a truncate operation with no timestamp.
+
+@wttest.skip_for_hook("nonstandalone", "timestamped truncate not supported for nonstandalone")
```

Comment:
The condition in `btcur_range_truncate` to block the fast truncate with `no_timestamp` can be removed and this test can be allowed to run in both `standalone` and `non-standalone` variants.

Reply:
@kommiharibabu Do you mean that we can remove this entire section here?

`#ifdef WT_STANDALONE_BUILD
    /*
     * All historical versions must be removed when a key is updated with no timestamp, but that
     * isn't possible in fast truncate operations. Disallow fast truncate in transactions configured
     * to commit without a timestamp (excluding logged tables as timestamps cannot be relevant to
     * them).
     */
    if (!F_ISSET(btree, WT_BTREE_LOGGED) && F_ISSET(session->txn, WT_TXN_TS_NOT_SET))
        WT_RET_MSG(session, EINVAL,
          "truncate operations may not yet be included in transactions that can commit without a "
          "timestamp. If your use case encounters this error, please reach out to the WiredTiger "
          "team");
#endif`

Reply:
Yes. By removing that section, we can achieve the same behavior for both standalone and non-standalone builds.